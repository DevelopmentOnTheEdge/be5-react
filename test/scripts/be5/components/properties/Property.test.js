import React from 'react';
import Property from '../../../../../src/scripts/be5/components/properties/property';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

const item = "/testName";
const itemName = item.substring(item.lastIndexOf("/")+1);
const itemValue = "testValue";

/*
    snapshot test
*/
it('simple property', () => {
	const itemMeta = {"displayName": "Simple property"};

	const component = renderer.create(
    <Property meta={itemMeta} name={itemName} value={itemValue} path={item}
    													  key={itemName} />
  );
  let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

it('property Boolean', () => {
    const itemMeta = {"displayName":"Boolean"};
    const item = "/agree";


    let component = renderer.create(
        <Property path={item} meta={itemMeta} name={itemName} key={itemName} value={"true"}/>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

/*
    shallow test
*/
it('call callback after click', () => {
	const itemMeta = {};

	const handle = jest.fn();

  const wrapper = shallow(
    <Property meta={itemMeta} name={itemName} value={itemValue} path={item}
        													  key={itemName} onChange={handle} />
  );

  wrapper.find('input').simulate('change', {target: {value: 'My new value'}});

  expect(handle.mock.calls.length).toBe(1);
  expect(handle.mock.calls[0]).toEqual(["/testName", "My new value"]);
  //TODO
  //expect(myMock.mock.calls[0]).toEqual(["/testName", true]);
});

it('call callback after set text', () => {
	const itemMeta = {};
	const itemValue = "testValue";

	const handle = jest.fn();

  const wrapper = shallow(
    <Property meta={itemMeta} name={itemName} value={itemValue} path={item}
        													  key={itemName} onChange={handle} />
  );

	wrapper.find('input').simulate('change', {target: {value: 'My new value'}});

  expect(handle.mock.calls.length).toBe(1);
  expect(handle.mock.calls[0]).toEqual(["/testName", "My new value"]);
});