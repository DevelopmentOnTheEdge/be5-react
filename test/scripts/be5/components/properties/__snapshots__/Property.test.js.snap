// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple property 1`] = `
<div
  className="form-group property col-xs-12 required"
>
  <label
    className="form-control-label"
    htmlFor="testNameField"
  >
    Simple property
  </label>
  <div
    className="controls"
  >
    <input
      className="form-control"
      disabled={undefined}
      id="testNameField"
      onChange={[Function]}
      placeholder={undefined}
      type="text"
      value="testValue"
    />
  </div>
</div>
`;
